### PR TITLE
Add PR number to build scans

### PR DESCRIPTION
**Pull request by gsmet**
The previous way of doing it didn't work: we are now extracting the value from the event.json file in the Groovy script.

### CODE REVIEW

File: .mvn/gradle-enterprise-custom-user-data.groovy [hunk 1]
```diff
@@ -17,11 +17,17 @@ if (System.env.GITHUB_ACTIONS) {
     buildScan.value('gh-actor', System.env.GITHUB_ACTOR)
     buildScan.value('gh-workflow', System.env.GITHUB_WORKFLOW)
 
+    if (System.env.GITHUB_EVENT_NAME == "pull_request" && System.env.GITHUB_EVENT_PATH != null) {
```
Comment (maxandersen) [inline .mvn/gradle-enterprise-custom-user-data.groovy:20]: don't we have cases where jobs are triggered by a pullrequest as a workflow request where this info won't exist?

**Review by maxandersen — APPROVED** [review]
yes, thats the right change and better what we have now!

+1 

my only comment is if we are able to capture PR for the cases where there are spinoff jobs but that can be improved in future.

File: .mvn/gradle-enterprise-custom-user-data.groovy (hunk 1 above)
Comment (gsmet) [reply .mvn/gradle-enterprise-custom-user-data.groovy:20]: I have no idea what you're saying :). Could you explain which case you are trying to cover?

Comment (gsmet) [reply .mvn/gradle-enterprise-custom-user-data.groovy:20]: We discussed it with Max, what I did is fine and this particular case has nothing to do with PR execution but withe event payloads sent to the bot webhook.

**Comment by gsmet** [comment]
@jprinet this should solve the issue with PR build scans not being properly tagged and I should be able to properly identify the build scans once this is in.

**Comment by geoand** [comment]
`VertxMDCTest` seems to be very flaky

**Pull request closed without merging**